docker/testnet/geth_setup/entry.sh

```diff
@@ -35,63 +35,99 @@ passkey_credential_id=${PASSKEY_CREDENTIAL_ID:-"9h5F3DgLSjSMdnVOadmhCw"}
 passkey_y=${PASSKEY_Y:-104273800132786176334597151467609377740095818152192999025225464410568038480397}
 
-entry_point_address="0x663F3ad617193148711d28f5334eE4Ed07016602"
-simple_account_factory_address="0x2E983A1Ba5e8b38AAAeC4B440B9dDcFBf72E15d1"
-passkey_account_factory_address="0xBC9129Dc0487fc2E169941C75aABC539f208fb01"
-verifying_paymaster_address="0xF6168876932289D073567f347121A267095f3DD6"
+##############################
+# Deploy periphery contracts #
+##############################
 
-# Deploy non-account abstraction contracts
-echo -e "\033[0;33m[Deploy non-account abstraction contracts]\033[0m"
+echo -e "\033[0;33m[Deploy periphery contracts]\033[0m"
 
 # Deploy Counter
 echo -e "\033[0;33m[Deploy Counter]\033[0m"
 forge create --rpc-url ${rpc_url} --private-key ${deployer_1_private_key} src/Counter.sol:Counter
 
-# Deploy account abstraction v0.6.0 contracts
+###############################################
+# Deploy account abstraction v0.6.0 contracts #
+###############################################
+
 echo -e "\033[0;33m[Deploy account abstraction v0.6.0 contracts]\033[0m"
 
-# Deploy EntryPoint
-echo -e "\033[0;33m[Deploy EntryPoint]\033[0m"
+entry_point_address_v0_6="0x663F3ad617193148711d28f5334eE4Ed07016602"
+simple_account_factory_address_v0_6="0x2E983A1Ba5e8b38AAAeC4B440B9dDcFBf72E15d1"
+simple_account_address_v0_6="0x1E684E8937774B00Ee2Ea562256f27a5c9D20d7c"
+passkey_account_factory_address_v0_6="0xBC9129Dc0487fc2E169941C75aABC539f208fb01"
+verifying_paymaster_address_v0_6="0xF6168876932289D073567f347121A267095f3DD6"
+
+# Deploy EntryPoint v0.6
+echo -e "\033[0;33m[Deploy EntryPoint v0.6]\033[0m"
 forge create --rpc-url ${rpc_url} --private-key ${deployer_2_private_key} lib/account-abstraction/0.6/contracts/core/EntryPoint.sol:EntryPoint
 
-# Deploy SimpleAccountFactory
-echo -e "\033[0;33m[Deploy SimpleAccountFactory]\033[0m"
-forge create --rpc-url ${rpc_url} --private-key ${deployer_2_private_key} lib/account-abstraction/0.6/contracts/samples/SimpleAccountFactory.sol:SimpleAccountFactory --constructor-args ${entry_point_address}
+# Deploy SimpleAccountFactory v0.6
+echo -e "\033[0;33m[Deploy SimpleAccountFactory v0.6]\033[0m"
+forge create --rpc-url ${rpc_url} --private-key ${deployer_2_private_key} lib/account-abstraction/0.6/contracts/samples/SimpleAccountFactory.sol:SimpleAccountFactory --constructor-args ${entry_point_address_v0_6}
+
+# Deploy SimpleAccount v0.6
+echo -e "\033[0;33m[Create SimpleAccount v0.6]\033[0m"
+cast send --rpc-url ${rpc_url} --private-key ${deployer_2_private_key} ${simple_account_factory_address_v0_6} "createAccount(address owner,uint256 salt)" ${operator_address} 0
+
+# Get SimpleAccount v0.6 address
+echo -e "\033[0;33m[Get SimpleAccount v0.6 address]\033[0m"
+echo $(cast call --rpc-url ${rpc_url} ${simple_account_factory_address_v0_6} "getAddress(address owner,uint256 salt)" ${operator_address} 0 | sed -r 's/^[.]*(0x)([0]{24}|)([0-9a-zA-Z]{40})[.]*$/\1\3/g')
```

Comment:
Since SimpleAccount and PasskeyAccount addresses can be obtained from the contract,
we might consider using variables to get their addresses without separate declarations to avoid unnecessary future modifications (the addresses declared here).

```shell
# ...
# simple_account_address_v0_6="0x1E684E8937774B00Ee2Ea562256f27a5c9D20d7c"
# ...

simple_account_address_v0_6=$(cast call --rpc-url ${rpc_url} ${simple_account_factory_address_v0_6} "getAddress(address owner,uint256 salt)" ${operator_address} 0 | sed -r 's/^[.]*(0x)([0]{24}|)([0-9a-zA-Z]{40})[.]*$/\1\3/g')
echo ${simple_account_address_v0_6}

# ...
# simple_account_address_v0_7="0xe569f1d8487239659C09b5cA1881320B5EbB0ab2"
# ...

simple_account_address_v0_7=$(cast call --rpc-url ${rpc_url} ${simple_account_factory_address_v0_7} "getAddress(address owner,uint256 salt)" ${operator_address} 0 | sed -r 's/^[.]*(0x)([0]{24}|)([0-9a-zA-Z]{40})[.]*$/\1\3/g')
echo ${simple_account_address_v0_7}
 # ...
```
What do you think?

Reply:
Since simple account would have fixed address, in my opinion it is more ideal to treat them just like other contracts, for easy reference their addresses at the top of the script.

And because passkey account accepts custom env, it is not likely to fix its address beforehand, so passkey account keeps the pattern as you mentioned.

Reply:
OK!